pkg/translator/prometheusremotewrite/helper.go

```diff
@@ -575,6 +604,44 @@ func addResourceTargetInfo(resource pcommon.Resource, settings Settings, timesta
 	addSample(tsMap, sample, labels, infoType)
 }
 
+func addScopeTargetInfo(
+	scope pcommon.InstrumentationScope,
+	resource pcommon.Resource,
+	settings Settings,
+	timestamp pcommon.Timestamp,
+	tsMap map[string]*prompb.TimeSeries,
+) {
+	if settings.DisableScopeInfo {
+		return
+	}
+	if scope.Attributes().Len() == 0 {
+		// If the scope doesn't have additional attributes, then otel_scope_info isn't useful.
```

Comment:
> It will contain the name and version of the instrumentation scope. Isn't this information per se useful?

It will contain the name and version of the instrumentation scope. Isn't this information per se useful? It means that there is at least a single metric being emitted in that scope.

Having `scope_info` in the configuration enabled and not generating the `otel_scope_info` metric is misleading and will confuse users.

Reply:
> Isn't this information per se useful? It means that there is at least a single metric being emitted in that scope.

If one was looking for metrics under a specific scope, they'd probably do something like a topk query on
```
__name__{otel_scope_name="w/e"}
``` 
In general, actual usage of `otel_scope_info` is most likely after knowing the `__name__` you're after then
```
<query> / on (otel_scope_name, otel_scope_version) group_left otel_scope_info
```
Though the query would return no data if `otel_scope_info` doesn't exist. So if the feature is enabled, it might make sense to always report it so queries that use the metric always return data regardless of whether or not the joined labels add novel information

Reply:
I pushed a commit that emitted the metric regardless.

But now I'm second guessing it. This has the potential to add a lot of active time series that won't be valuable the majority of the time. The collector could receive metrics from many languages and only PHP and rust even have the potential to make this metric add novel information. And the query with the join tends to be a rather advanced use case (so advanced that documentation doesn't exist around how to properly use `otel_scope_info` let alone `target_info`) Additionally, not emitting the metric if it doesn't have attributes would align with the implementation of `target_info`: https://github.com/blakeroberts-wk/opentelemetry-collector-contrib/blob/3f146d90bd9c622c33359c71fcae89162a263f2c/pkg/translator/prometheusremotewrite/helper.go#L589-L591